src/views/states/list/hooks/useStateFilters.ts

```diff
@@ -38,6 +43,36 @@ export const useStateSearchFilters = (): RowSearchFilter<V1beta1NodeNetworkState
       filterGroupName: t('MAC address'),
       placeholder: t('Search by MAC address...'),
     },
+    {
+      type: FILTER_TYPES.LLDP_NAME,
+      filter: (searchText, obj) => {
+        const searchLLDPName = searchText?.selected?.[0];
+        if (isEmpty(searchLLDPName)) return true;
+
+        const interfaces = obj?.status?.currentState
+          ?.interfaces as NodeNetworkConfigurationInterface[];
+
+        return interfaces?.some((iface) => searchInterfaceByLLDPName(searchLLDPName, iface));
+      },
+      filterGroupName: t('LLDP VLAN name'),
+      placeholder: t('Search by VLAN name...'),
+    },
+    {
+      type: FILTER_TYPES.LLDP_SYSTEM_NAME,
+      filter: (searchText, obj) => {
+        const searchSystemName = searchText?.selected?.[0];
+        if (isEmpty(searchSystemName)) return true;
+
+        const interfaces = obj?.status?.currentState
+          ?.interfaces as NodeNetworkConfigurationInterface[];
```

Comment:
Here I can see some repeating code. Maybe we could create some getter or some helper function to simply get what we want from the object? `getInterfaces(obj)` maybe? Then you could use it where needed and the code could look nicer.

Reply:
good point